src/libraries/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Principal.cs

```diff
@@ -124,7 +124,7 @@ public string SamAccountName
             set
             {
                 if (null == value || 0 == value.Length)
-                    throw new ArgumentNullException(PropertyNames.PrincipalSamAccountName);
+                    throw new ArgumentNullException(nameof(value));
```

Comment:
I don't know that we should be making this change.  Have we decided in general that the "right" argument name for property setters is `value` rather than the name of the property?

Reply:
In my mind, it is more useful to use the name of the property. However I have asked this in the past and there did not seem to be consensus.

Reply:
We didn't exactly discussed about that, but for now `value` is the "right" one, open for suggestions CC @terrajobst @bartonjs

Reply:
> Have we decided in general that the "right" argument name for property setters is value rather than the name of the property?

So much so that it's a design guideline: "DO use value for the name of the implicit value parameter of property
setters." (Framework Design Guidelines, 2nd edition, 7.3.4, pp 236).

Reply:
> So much so that it's a design guideline: "DO use value for the name of the implicit value parameter of property setters."

Isn't that about the name of the parameter as it shows up in metadata, rather than about the name that's used in the exception?

My main concern here is that properties are generally small, which means properties are more likely to be inlined, and if they are, an ArgumentException showing up with "value" and with the caller on the call stack isn't nearly as debuggable.

If we don't think that's an issue, ok.

Reply:
Surely "'PrincipalSamAccountName' is null" is at least as useful, and typically more useful, than "'value' is null". Why would we want to use value?

Reply:
Allowing name of the `parameter name` sounds reasonable, I guess we want to allow both, not only `parameter name`, I assume `value` also can be used

Reply:
That guideline is in the section of using ArgumentException exceptions correctly, so it's exactly this case.  The message can be customized to whatever seems the most relevant, but the guidelines have said for years that the paramName for an ArgumentException-family exception from a property setter should be "value".

This matches Immo's suggestion that the paramName always be something that tooling can highlight the parameter index on, and that a customized message is used when paramName and the exception type name aren't sufficient to fix the problem,

Reply:
Thanks @bartonjs! leaving as is then